tutorials/how_to_test_in_Flutter.md

```diff
@@ -0,0 +1,217 @@
+
+### Standard classes
+A test for a standard class looks like this:
+``` Dart
+ test('Test desctription',(){
+
+ });
+```
+If the code does not run in sequence you should use a async test, that looks this way:
+``` Dart
+ test('Test description', async((DoneFn done){
+     done();
+ }));
+```
+In the normal test, the test will finish when all the code are executed, while an async test on a standard class will finish when the done() function is called. If you do an async test, you should always remember to call the done() function in standard classes, because else test will continue until timeout and then fail. The done() function always have to be in inside the async function, not outside.
+
+``` Dart
+import 'package:<All the different imports>';
+
+void main() {
+    SomeClass a;
+    SomeOtherClass b;
+    BlocType bloc;
+
+    setUp((){
+        a = <Some value>;
+        b = <Some other value>;
+        bloc = bloc(some API);
+    });
+    
+    test('This is a standard test',(){
+        a.someMethod(b);
+        expect(b.something, 2);
+        expect(a.something, 3);
+    });
+
+    test('This is an async test',async((DoneFn done){
+        bloc.something.skip(1).listen((SomeModel model){
+            expect(model.content, value);
+            expect(model.otherContent, otherValue);
+            done();
+        });
+        bloc.setSomething(value);
+    });
+}
+```
+In the async test, the listen function is defined first. When the set function on Something is called, the listen function is notified and all the expect functions are called. 
+
+## Widgets and screens
+The widget and screen tests separate from the class tests. All tests are still in the main() function and you can still use a setup function, but instead of the normal tests you use test widgets. 
+``` Dart
+    testWidgets('Name',(WidgetTester tester) async {
+        await tester.<Something to wait for>;
+    });
+```
+Inside the testWidget you use a WidgetTester. This is used to interact with widgets in the test environment. When you use the WidgetTester for actions, you should use the await key word before. 
+
+In standard classes you use the done() function when running an async test. This is not done in testWidgets. However, if you use a listener in your expect, you use something called a completer.
+``` Dart
+    testWidgets('Name',(WidgetTester tester) async {
+        final Completer<datatypeA> done = Completer<datatypeA>();
+
+        bloc.someBloc.listen((datatypeA) async{
+            await tester.pump();
+            expect(success, true);
+            done.complete();
+        });
+        await done.future;
+    });
+```
+
+There are some basic functions we use a lot on the WidgetTester.
+
+The first is pumpWidget. This acts by setting up a screen. You can write it like this:
+``` Dart
+await tester.pumpWidget(MaterialApp(home:SomeScreen()));
```

Comment:
properply needs to explain more in depth when to use these

Reply:
Tried to add